tests/unit/amo/reducers/test_addonsByAuthors.js

```diff
@@ -83,59 +101,80 @@ describe(__filename, () => {
 
       const previousState = reducer(undefined, loadAddonsByAuthors({
         addons: [fakeAddon],
+        authorNames: [fakeAddon.authors[0].username],
         forAddonSlug,
       }));
+
       expect(previousState.byAddonSlug)
         .toEqual({ 'addon-slug': [fakeAddon.id] });
 
       const state = reducer(previousState, fetchAddonsByAuthors({
-        authors: ['author2'],
+        authorNames: ['author2'],
         addonType: ADDON_TYPE_THEME,
         errorHandlerId: 'error-handler-id',
       }));
 
       expect(state.byAddonSlug).toEqual({ 'addon-slug': [fakeAddon.id] });
     });
 
-    it('resets the loaded add-ons', () => {
+    it('resets the loaded add-ons for the slug', () => {
       const forAddonSlug = 'addon-slug';
 
-      const previousState = reducer(undefined, loadAddonsByAuthors({
+      const firstState = reducer(undefined, loadAddonsByAuthors({
         addons: [fakeAddon],
+        authorNames: [fakeAddon.authors[0].username],
         forAddonSlug,
       }));
-      expect(previousState.byAddonSlug)
-        .toEqual({ 'addon-slug': [fakeAddon.id] });
 
-      const state = reducer(previousState, fetchAddonsByAuthors({
-        authors: ['author1'],
+      expect(firstState.byAddonSlug).toEqual({ 'addon-slug': [fakeAddon.id] });
+
+      const state = reducer(firstState, fetchAddonsByAuthors({
+        authorNames: ['author1'],
         addonType: ADDON_TYPE_THEME,
         errorHandlerId: 'error-handler-id',
         forAddonSlug,
       }));
 
       expect(state.byAddonSlug).toMatchObject({ 'addon-slug': undefined });
-      expect(state.byUsername).toMatchObject({ author1: undefined });
+      // It should keep the add-ons by username so they can be added to
+      // as is done on the UserProfile page.
+      expect(state.byUsername).toMatchObject(firstState.byUsername);
+    });
+
+    it('sets the loading state for authorNames + addonType on fetch', () => {
```

Comment:
We're missing a test that checks that `forAuthorNamesAndAddonType` is set to `null` on fetch.

Reply:
I don't understand, when would `forAuthorNamesAndAddonType` be `null`? Do you mean have an empty array? This comment isn't clear, but I can add a test for an empty authorName array.

Reply:
Actually that would never happen so I don't understand this comment.

Reply:
Maybe I didn't communicate it well, it's not the entire `byAuthorNamesAndAddonType` object that gets set to null, but rather the particular key in question. It happens right here: https://github.com/mozilla/addons-frontend/pull/4611/files#diff-557fb278613a33409aa465086be9abb7R186, and we don't seem to have a test that checks that this is set to `null` upon fetch.

Reply:
Ah, gotcha. I've added it.